src/datasets/arrow_dataset.py

```diff
@@ -1737,6 +1736,8 @@ def filter(
             suffix_template=suffix_template,
             new_fingerprint=new_fingerprint,
         )
```

Comment:
The `map_function` that is provided returns all the examples that return `True` when passed to `function`.
Therefore what the `map` returns here is actually the filtered dataset already.

You can get a dataset containing the actual mask by replacing the `map_function` passed to map.
Maybe replace it with a function `compute_mask` that returns a dict like `{"mask": [True, False, False, ...]}`
given a batch of examples.

You'd also need to set `remove_columns=self.column_names` to have only the column "mask" in the ouput dataset (otherwise the other columns of the dataset are kept as well)